node/lib/util/stitch_util.js

```diff
@@ -778,8 +778,15 @@ exports.readConvertedCommits = co.wrap(function *(repo) {
 
     const result =
                  yield BulkNotesUtil.readNotes(repo, exports.convertedNoteRef);
-    for (let key in result) {
-        result[key] = exports.readConvertedContent(result[key]);
+    for (let key of Object.keys(result)) {
+        const sha = exports.readConvertedContent(result[key]);
+        try {
+            yield repo.getCommit(sha);
+        } catch (e) {
+            // We have the note but not the commit, delete from cache
+            delete result[key];
```

Comment:
the outside loops over keys and inside the loop, it changes the key set, this pattern reminds me of the ConcurrentModificationException in java. Is it too expensive to return a new object instead of modifying the existing object?

Reply:
No, it loops over Object.keys(result). That's computed first (not lazily).